actions/create-update-pr/create-update-pr.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+if ! [[ $BRANCH_NAME ]]; then
+    echo "BRANCH_NAME variable is required"
+    exit 1
+fi
+if ! [[ $BASE_BRANCH ]]; then
+    echo "BASE_BRANCH variable is required"
+    exit 1
+fi
+if ! [[ $COMMIT_MESSSAGE ]]; then
+    echo "COMMIT_MESSSAGE variable is required"
+    exit 1
+fi
+if ! [[ $PR_TITLE ]]; then
+    echo "PR_TITLE variable is required"
+    exit 1
+fi
+if ! [[ $ACTION_URL ]]; then
+    echo "ACTION_URL variable is required"
+    exit 1
+fi
+if ! [[ $RUN_ID ]]; then
+    echo "RUN_ID variable is required"
+    exit 1
+fi
+
+# Fetch
+git fetch
+
+# Create Branch
+git stash
+git switch "$BRANCH_NAME" || git switch -c "$BRANCH_NAME"
+
+# Push if Update
+if ! [[ $(git ls-remote --exit-code --heads origin refs/heads/"$BRANCH_NAME") &&  $(git diff --exit-code "$BRANCH_NAME" origin/"$BRANCH_NAME") ]]; then
+    echo "Changes detected. Creating commit and pushing"
+    git commit -a -m "$COMMIT_MESSSAGE"
+    git push -f --set-upstream origin "$BRANCH_NAME"
```

Comment:
Will this still work if you do `--force-with-lease` instead? If so I'd prefer that. Also is it guaranteed that `BRANCH_NAME` won't be main or master?

Reply:
Not exactly sure what force with lease do but I'll test it

The action guarantees that branch name cannot be master, but you could set it to master.

Reply:
Basically `--force-with-lease` will fail if any changes were made to the branch by someone else, while `--force` will always overwrite the changes. Usually you want to use `--force-with-lease` to avoid overwriting someone else's changes. In this example there might not be much of a difference, but let's err on the side of caution with `--force-with-lease`

Reply:
Changed and seems to work. When only action bot changes the PR, it still pushes